src/dwn/models/record.js

```diff
@@ -0,0 +1,244 @@
+import { DataStream, DwnConstant, Encoder } from '@tbd54566975/dwn-sdk-js';
+import { ReadableWebToNodeStream } from 'readable-web-to-node-stream';
+
+import { isReadableWebStream } from '../dwn-utils.js';
+
+export class Record {
+  #dwn;
+
+  #author;
+  #contextId;
+  #descriptor;
+  #recordId;
+  #target;
+
+  #encodedData = null;
+  #isFrozen = false;
+  #readableStream = null;
+
+  constructor(dwn, options = { }) {
+    this.#dwn = dwn;
+
+    // RecordsWriteMessage properties.
+    const { author, contextId = undefined, descriptor, recordId = null, target } = options;
+    this.#contextId = contextId;
+    if (descriptor?.data) delete descriptor.data;
+    this.#descriptor = descriptor ?? { };
+    this.#recordId = recordId;
+    
+    // Store the target and author DIDs that were used to create the message to use for subsequent reads, etc.
+    this.#author = author;
+    this.#target = target;
+    
+    // If the record `dataSize is less than the DwnConstant.maxDataSizeAllowedToBeEncoded value,
+    // then an `encodedData` property will be present.
+    this.#encodedData = options?.encodedData ?? null;
+
+    // If the record was created from a RecordsRead reply then it will have a `data` property.
+    if (options?.data) {
+      // this.#readableStream = isReadableWebStream(options.data) ? toIsomorphicReadableStream(options.data) : options.data;
+      this.#readableStream = isReadableWebStream(options.data) ? new ReadableWebToNodeStream(options.data) : options.data;
+    }
+  }
+
+  // Mutable Web5 Record Class properties.
+  get author() { return this.#author; }
+  get isFrozen() { return this.#isFrozen; }
```

Comment:
you say that this is mutable but i dont see a `set isFrozen`?  is that actually mutable then?

Reply:
It isn't mutable directly.  It would be changed as a result of an undelete.